From 9402dfdd718e0a1c0cbdcea5a4a475ee9572b38c Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sat, 26 Oct 2024 20:13:34 +1100
Subject: [PATCH 4/5] don't declare RowMajorStrg as extern on windows

on platforms with HAS_ATTRIBUTE_WEAK_SUPPORT, this presumably plays a role in
the linker shenanigans with cblas_xerbla, but on windows it just blows up
---
 CBLAS/testing/c_c2chke.c | 2 +-
 CBLAS/testing/c_c3chke.c | 2 +-
 CBLAS/testing/c_d2chke.c | 2 +-
 CBLAS/testing/c_d3chke.c | 2 +-
 CBLAS/testing/c_s2chke.c | 2 +-
 CBLAS/testing/c_s3chke.c | 2 +-
 CBLAS/testing/c_z2chke.c | 2 +-
 CBLAS/testing/c_z3chke.c | 2 +-
 8 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/CBLAS/testing/c_c2chke.c b/CBLAS/testing/c_c2chke.c
index 28b771980..9a26ece56 100644
--- a/CBLAS/testing/c_c2chke.c
+++ b/CBLAS/testing/c_c2chke.c
@@ -33,7 +33,7 @@ void F77_c2chke(char *rout) {
           BETA[2]  = {0.0,0.0},
           RALPHA = 0.0;
    extern int cblas_info, cblas_lerr, cblas_ok;
-   extern int RowMajorStrg;
+   int RowMajorStrg = TRUE;
    extern char *cblas_rout;
 
    if (link_xerbla) /* call these first to link */
diff --git a/CBLAS/testing/c_c3chke.c b/CBLAS/testing/c_c3chke.c
index 1be0c3fd1..29cf86ee8 100644
--- a/CBLAS/testing/c_c3chke.c
+++ b/CBLAS/testing/c_c3chke.c
@@ -33,7 +33,7 @@ void  F77_c3chke(char *  rout) {
            BETA[2]  = {0.0,0.0},
            RALPHA   = 0.0, RBETA = 0.0;
    extern int cblas_info, cblas_lerr, cblas_ok;
-   extern int RowMajorStrg;
+   int RowMajorStrg = TRUE;
    extern char *cblas_rout;
 
    cblas_ok = TRUE ;
diff --git a/CBLAS/testing/c_d2chke.c b/CBLAS/testing/c_d2chke.c
index d989811d2..4d6c122f0 100644
--- a/CBLAS/testing/c_d2chke.c
+++ b/CBLAS/testing/c_d2chke.c
@@ -31,7 +31,7 @@ void F77_d2chke(char *rout) {
           Y[2] = {0.0,0.0},
           ALPHA=0.0, BETA=0.0;
    extern int cblas_info, cblas_lerr, cblas_ok;
-   extern int RowMajorStrg;
+   int RowMajorStrg = TRUE;
    extern char *cblas_rout;
 
    if (link_xerbla) /* call these first to link */
diff --git a/CBLAS/testing/c_d3chke.c b/CBLAS/testing/c_d3chke.c
index e41901e79..bf155684e 100644
--- a/CBLAS/testing/c_d3chke.c
+++ b/CBLAS/testing/c_d3chke.c
@@ -31,7 +31,7 @@ void F77_d3chke(char *rout) {
           C[2] = {0.0,0.0},
           ALPHA=0.0, BETA=0.0;
    extern int cblas_info, cblas_lerr, cblas_ok;
-   extern int RowMajorStrg;
+   int RowMajorStrg = TRUE;
    extern char *cblas_rout;
 
    if (link_xerbla) /* call these first to link */
diff --git a/CBLAS/testing/c_s2chke.c b/CBLAS/testing/c_s2chke.c
index 63f167152..11c6af7e1 100644
--- a/CBLAS/testing/c_s2chke.c
+++ b/CBLAS/testing/c_s2chke.c
@@ -31,7 +31,7 @@ void F77_s2chke(char *rout) {
           Y[2] = {0.0,0.0},
           ALPHA=0.0, BETA=0.0;
    extern int cblas_info, cblas_lerr, cblas_ok;
-   extern int RowMajorStrg;
+   int RowMajorStrg = TRUE;
    extern char *cblas_rout;
 
    if (link_xerbla) /* call these first to link */
diff --git a/CBLAS/testing/c_s3chke.c b/CBLAS/testing/c_s3chke.c
index 1872f005c..bd09a8041 100644
--- a/CBLAS/testing/c_s3chke.c
+++ b/CBLAS/testing/c_s3chke.c
@@ -31,7 +31,7 @@ void F77_s3chke(char *rout) {
           C[2] = {0.0,0.0},
           ALPHA=0.0, BETA=0.0;
    extern int cblas_info, cblas_lerr, cblas_ok;
-   extern int RowMajorStrg;
+   int RowMajorStrg = TRUE;
    extern char *cblas_rout;
 
    if (link_xerbla) /* call these first to link */
diff --git a/CBLAS/testing/c_z2chke.c b/CBLAS/testing/c_z2chke.c
index d51c7c267..8a888be95 100644
--- a/CBLAS/testing/c_z2chke.c
+++ b/CBLAS/testing/c_z2chke.c
@@ -33,7 +33,7 @@ void F77_z2chke(char *rout) {
           BETA[2]  = {0.0,0.0},
           RALPHA = 0.0;
    extern int cblas_info, cblas_lerr, cblas_ok;
-   extern int RowMajorStrg;
+   int RowMajorStrg = TRUE;
    extern char *cblas_rout;
 
    if (link_xerbla) /* call these first to link */
diff --git a/CBLAS/testing/c_z3chke.c b/CBLAS/testing/c_z3chke.c
index 10078a103..bf8a6f9b3 100644
--- a/CBLAS/testing/c_z3chke.c
+++ b/CBLAS/testing/c_z3chke.c
@@ -33,7 +33,7 @@ void  F77_z3chke(char *  rout) {
            BETA[2]  = {0.0,0.0},
            RALPHA   = 0.0, RBETA = 0.0;
    extern int cblas_info, cblas_lerr, cblas_ok;
-   extern int RowMajorStrg;
+   int RowMajorStrg = TRUE;
    extern char *cblas_rout;
 
    cblas_ok = TRUE ;
